Implemented Clone for `NamedField`s

Razican opened the pull request:
This implements `std::clone::Clone` for `NamedField`s, if the inner type is `Clone`. This allows chaining with #1, to easily do transformations in returned rows.

weiznich commented:
Is there any reason that requires a explicit `Clone` impl and cannot use the derive?

Razican commented:
The reason was I didn't know the derive implementation was intelligent enough to do bounds checking xD
Fixed, and also implemented it for `DynamicRow`.
